### DISCUSSION TITLE
program arg

### DISCUSSION THREAD

**Discussion by RamadhanAmizudin** [BishopFox/cloudfox#24] [Q&A] [answered]
Hi, based on prerequisite stated in readme, "Supports AWS profiles, AWS environment variables, or metadata retrieval (on an ec2 instance) " 
I do know how to pass based on AWS profiles, but how to pass key/secret with env var, or instruct cloudfox to do metadata retrieval?

Unless i missing something, but i found nothing on wiki. hope you can assist on this matter

thank you

**Comment by sethsec-bf** [comment, marked as answer]
Hi @RamadhanAmizudin. for metadata service, just run cloudfox **without** the -p parameter and it will use the metadata service and retrieve the creds for you as long as their is an IAM role attached to the instance.  for enmvironment variables it uses the same evironment variables as the AWS CLI: 

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

```
$ export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
$ export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
$ export AWS_DEFAULT_REGION=us-west-2
``` 

Let me know if either of these don't work as expected.
